docs/proposals/002-api-proposal/proposal.md

```diff
@@ -0,0 +1,254 @@
+- Managing rollout of adapters
+
+### Axioms 
+
+The API design is based on these axioms:
+
+- Pools of shared compute should be *discrete* for scheduling to properly work
+- Pod-level scheduling should not be handled by a high-level gateway 
+- Simple use cases should be simple to define (or are implicitly defined via reasonable defaults)
+- This solution should be composable with other Gateway solutions and flexible to fit customer needs
+- The MVP will heavily assume requests are done using the OpenAI spec, but open to extension in the future
+
+The [PoC](https://youtu.be/NUBZg_uqqXk?si=v681EeYdGUGEVqQQ&t=1458) was focused on lower-level scheduling. And the API follows that similar logic, which lead to the proposal of the **BackendPool**.
+
+### BackendPool
+
+The BackendPool at its core is a logical grouping of compute, expressed in the form of Pods (typically model servers), akin to a K8s Service. The BackendPool would deploy its own routing, and offer administrative configuration to the Platform Admin. 
+
+ It is expected for the BackendPool to:
+ - Enforce fair consumption of resources across competing use cases
+ - Efficiently route requests across shared compute (as displayed by the PoC)
+ 
+It is _not_ expected for the BackendPool to:
+ - Enforce any common set of adapters or base models are available on the Pods
+ - Manage Deployments of Pods within the Pool
+ - Manage Pod lifecycle of pods within the pool 
+
+Additionally, any Pod that seeks to join a BackendPool would need to support a protocol, defined by LLM Instance Gateway, to ensure the Pool has adequate information to intelligently route requests.
+
+### LLMUseCase
+
+An LLMUseCase allows the UseCaseOwner to define:
+- Which LoRA adapter(s) to consume 
+  - LLMUseCase allows for traffic splitting between adapters _in the same pool_ to allow for new LoRA adapter versions to be easily rolled out 
+- SLO objectives for the UseCAse
+- The Pools this UseCase is relevant to 
+
+### Spec
+
+**BackendPool**
+```golang
+// A grouping of model serving instances that are expected to serve the same model(s) and speak the same API protocol
+// The LLMBackendPool is also the scope for enforcing priority and fairness across different use cases.
+// When used with the Gateway API, the LLMBackendPool can serve as the BackendRefs of an HTTPRoute.
+// Generally the LLMBackendPool is owned by the "Inference Platform Admin" persona.
+type LLMBackendPool struct {
+        metav1.ObjectMeta
+        metav1.TypeMeta
+        Spec LLMBackendSpec
+}
+
+type LLMBackendPoolSpec struct {
+        // Pod selector, similar to k8s Service.
+        Selector map[string]string `json:"selector,omitempty"`
+        // Allows names within the `model` param that don't have an explicitly defined UseCase to pass through. Defaults to `false`.
+        AllowUndefinedModels bool
+        // Admin-defined minimum objective value that can be requested, will cause an error in LLMUseCase if limit is broken.
```

Comment:
What's the difference from `LLMUseCaseRule.objective`?

Reply:
Assuming we are talking about `MinTPOT` This was an attempt to show what administrative control might look like. This lets a platform admin declare policy, stating that no objective can ask for a TPOT below this configured value.